src/matrix.c

```diff
@@ -93,31 +94,35 @@ int ir_matrix_free_lua(lua_State *L) {
 }
 
 int ir_matrix_from_lua(lua_State *L) {
+    int argc;
 	float temp;
 	// mat4 result;
     // NOTE: For potential windows support we'd need _aligned_malloc instead ~FabricatorZayac
     mat4 *result = MAT4_ALLOC();
 
-	if (lua_gettop(L) == 1) {
-		lua_pop(L, lua_gettop(L));
-		return 0;
+    argc = lua_gettop(L);
+	if (argc != 1) {
+		lua_pop(L, argc);
+        return ir_push_error_lua(L, "Expected 1 argument, provided %d", argc);
```

Comment:
Not sure how I didn't catch this before, but this would prevent ir.mat.from from working at all!